libs/zedUpload/azureutil/azure.go

```diff
@@ -311,9 +324,17 @@ func DownloadAzureBlob(accountURL, accountName, accountKey, containerName, remot
 					progressReceiver(progress)
 					progressLock.Unlock()
 				})
-			_, err = io.Copy(newSectionWriter(file, chunkStart+offset, count-offset, part), body)
-			body.Close()
-			return err
+			// we get buffer from the pool to not allocate
+			// new buffer per every chunk inside io.copyBuffer
```

Comment:
capitalization nit; io.CopyBuffer

Reply:
Actually it is about unexported [io.copyBuffer](https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/io/io.go;drc=2580d0e08d5e9f979b943758d3c49877fb2324cb;l=404) function.

Reply:
Yeah, @giggsoff explained it to me. It is about reusing buffers.

Reply:
ok what wasn't not at all clear to me. Maybe saying "the internal io.copyBuffer".